File: tests/fixtures/treesitter/python/sample.py
"""Baseline fixture for tree-sitter Python parsing.

Covers: class with @decorator (preceding sibling), async def, PEP 695 type alias
(Python 3.12+ — `type X = ...`), import capture, method inside class.
"""
from typing import Optional

type OrderId = str  # PEP 695 — tree-sitter may capture as 'module' or 'type_alias'


@dataclass
class Order:
    id: str
    total: int = 0


@app.get("/orders")
async def get_orders() -> list[Order]:
    return []


@app.post("/orders")
def create_order(order: Order) -> Order:
    return order

File: tests/fixtures/treesitter/java/Sample.java
// Baseline fixture for tree-sitter Java parsing.
// Covers: class + @RestController, @GetMapping method, Java 16 record,
// @interface annotation type, import capture, generic method.

import org.springframework.web.bind.annotation.GetMapping;

@RestController
public class OrderController {

  @GetMapping("/orders/{id}")
  public Order getOrder(@PathVariable String id) {
    return new Order(id, 0);
  }

  public <T> List<T> findAll(Class<T> type) {
    return List.of();
  }
}

public record OrderRecord(String id, int total) {}

public @interface Validated {
  String message() default "";
}